gapis/resolve/dependencygraph2/README.md

```diff
@@ -0,0 +1,166 @@
+# Dependency graph
+
+The dependency graph tracks dependencies between the commands of a capture.
+
+> This doc is an overview of the dependency graph internals. As the code
+> evolves, the few code snippets used here may not match the latest version,
+> but should still help to understand the high-level approach.
+
+In general, Command B depends on command A if command A has some side-effect to
+the API state or some resource (image, buffer...) memory that impacts command B.
+
+To take a simple GL example:
+
+```
+glClearColor(1, 0, 0, 1);     // Set clear color to red
```

Comment:
Should we give a Vulkan example instead?

Reply:
done